.github/workflows/rust.yml

```diff
@@ -11,17 +11,11 @@ env:
   RUST_BACKTRACE: 1
 
 jobs:
-  check:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    - name: Format
-      run: cargo fmt --all -- --check
-    - name: Clippy
-      run: cargo clippy --all-targets --all-features -- -D warnings
-
-  test:
-    runs-on: ubuntu-latest
+  build-and-test:
```

Comment:
We've set it up to only run checks on Ubuntu for a quicker CI; don't merge them back. There's no need to run `fmt` and `clippy` on multiple platforms.